### PR TITLE
Refacto geosys.py into separate modules

### PR DESCRIPTION
**Pull request by Abdessamed-geosys**

### CODE REVIEW

File: geosyspy/geosys.py [hunk 1]
```diff
@@ -257,24 +126,24 @@ def get_time_series(self, polygon, start_date, end_date, collection, indicators)
         Raises:
             ValueError: The collection doesn't exist
         """
-
-        if collection in self.list_collection_weather:
+        # if collection in self.list_collection_weather:
```
Comment (qhl-geosys) [inline geosyspy/geosys.py]: From my PoV commented code is useless because in 99% cases it'll become dead code.

File: geosyspy/geosys.py [hunk 2]
```diff
@@ -658,29 +528,30 @@ def get_coordinates_by_pixel(raster):
...
-    def __get_weather(self, polygon, start_date, end_date, weather_type, fields):
+    def __get_weather(self, polygon, start_date, end_date, collection, fields):
         """Returns the weather data as a pandas dataframe.
 
         Args:
             polygon : A string representing a polygon.
             start_date : A datetime object representing the start date of the date interval the user wants to filter on.
             end_date : A datetime object representing the final date of the date interval the user wants to filter on.
-            weather_type : A string representing the collection ["HISTORICAL_DAILY", "FORECAST_DAILY", "FORECAST_HOURLY"]
+            collection : A string representing the collection ["HISTORICAL_DAILY", "FORECAST_DAILY", "FORECAST_HOURLY"] //
```
Comment (qhl-geosys) [inline geosyspy/geosys.py]: // ? at the end of the line 538

File: geosyspy/geosys.py [hunk 3]
```diff
@@ -658,29 +528,30 @@ def get_coordinates_by_pixel(raster):
...
         """
 
-        allowed_weather_types = [
-            "HISTORICAL_DAILY",
-            "FORECAST_DAILY",
-            "FORECAST_HOURLY",
-        ]
-        if weather_type not in allowed_weather_types:
-            raise ValueError(f"weather_type should be either {allowed_weather_types}")
+        # allowed_weather_types = [
```
Comment (qhl-geosys) [inline geosyspy/geosys.py]: dead code

File: setup.py [hunk 4]
```diff
@@ -25,5 +25,5 @@
     packages=["geosyspy"],
     include_package_data=True,
     data_files=[('', ['VERSION.txt'])],
-    install_requires=["requests", "requests-oauthlib", "oauthlib", "scipy", "pandas==1.3.5", "shapely", "rasterio", "xarray"]
+    install_requires=["requests", "requests-oauthlib", "oauthlib", "scipy", "pandas>=1.3.5", "shapely", "rasterio", "xarray"]
```
Comment (qhl-geosys) [inline setup.py]: Why has it been changed ?

File: geosyspy/geosys.py (hunk 1 above)
Comment (Abdessamed-B) [reply geosyspy/geosys.py]: I totally agree, got rid of unnecessary comment!

File: geosyspy/geosys.py (hunk 2 above)
Comment (Abdessamed-B) [reply geosyspy/geosys.py]: fixed it!

File: geosyspy/geosys.py (hunk 3 above)
Comment (Abdessamed-B) [reply geosyspy/geosys.py]: Deleted the dead code!

File: examples.ipynb
Comment (simon-geosys) [inline examples.ipynb:1]: Can you clear the output before commiting this file so we only get the changes in the cells ?

File: geosyspy/utils/client.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            return func(self, *args, **kwargs)
+        except TokenExpiredError:
+            self.access_token = self.__client_oauth.get_refresh_token()
+            return func(self, *args, **kwargs)
+
+    return wrapper
+
+
+class PlatformApiClient:
+    access_token: str
```
Comment (simon-geosys) [inline geosyspy/utils/client.py]: i don't think we need to keep all this info as class field, it doesn't seem to be used after being passed to the `OAuth2Api` class

File: setup.py (hunk 4 above)
Comment (Abdessamed-B) [reply setup.py]: This change was not meant to be pushed, now, it is canceled!

File: requirements.txt [hunk 6]
```diff
@@ -5,4 +5,5 @@ scipy
 pandas==1.3.5
 shapely
 rasterio
-xarray
\ No newline at end of file
+xarray
+requests-mock
```
Comment (simon-geosys) [inline requirements.txt]: Remove from requirements.txt, to keep only in requirements-dev.txt

File: geosyspy/utils/constants.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class GeosysApiEndpoints(Enum):
+    """
+    Available Geosys APIs Endpoints
+    """
+    MASTER_DATA_MANAGEMENT_ENDPOINT = "master-data-management/v6/seasonfields",
+    VTS_ENDPOINT = "vegetation-time-series/v1/season-fields",
+    VTS_BY_PIXEL_ENDPOINT = "vegetation-time-series/v1/season-fields/pixels",
+    FLM_CATALOG_IMAGERY = (
```
Comment (simon-geosys) [inline geosyspy/utils/constants.py]: Tuple to be removed

Comment (Abdessamed-B) [reply geosyspy/utils/constants.py]: Done!

File: requirements.txt (hunk 6 above)
Comment (Abdessamed-B) [reply requirements.txt]: Done!

File: geosyspy/utils/client.py (hunk 5 above)
Comment (Abdessamed-B) [reply geosyspy/utils/client.py]: Removed unused class attributes!

File: examples.ipynb
Comment (Abdessamed-B) [reply examples.ipynb:1]: Yes, now Jupyter notebook file has no cache saved :)

**Pull request merged**